### PR TITLE
add support for multiple eval sets (including n = 0)

### LINKED ISSUE #217: Support Multiple Validation Sets
Should be able to plug in an arbitrary number of validation sets

### PR DESCRIPTION
**Pull request by dlwh**
Fixes #217 #360

### CODE REVIEW

File: src/levanter/callbacks.py [hunk 1]
```diff
@@ -23,11 +23,16 @@
 logger = logging.getLogger(__name__)
 
 
-def eval_loss_loop(loss_fn, model, dataset, max_batches: Optional[int] = None):
+def eval_loss_loop(loss_fn, model, dataset, max_batches: Optional[int] = None, name: Optional[str] = None):
     total_loss = 0.0
     n = 0
 
-    pbar = tqdm(dataset, desc="eval", position=1, leave=False)
+    if name is not None:
```
Comment (Ivan-Zhou) [inline src/levanter/callbacks.py:30]: Not sure if I capture the logic correctly, but based on [here](https://github.com/stanford-crfm/levanter/blob/1f597337c079e7960ac3ac6ba9f1b33d3c1d6030/src/levanter/data/text.py#L567), it seems that for single dataset, the eval set name will be `""` instead of None, right?

File: config/gpt2_nano_mixture.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+data:
+  configs:
+    wikitext:
+      id: dlwh/wikitext_103_detokenized
+    w2:
+      id: dlwh/wikitext_103_detokenized
```
Comment (Ivan-Zhou) [inline config/gpt2_nano_mixture.yaml:6]: If we use different names, this will make two eval loss curves, despite that these are the same datasets. Nice trick.

File: src/levanter/callbacks.py (hunk 1 above)
Comment (dlwh) [reply src/levanter/callbacks.py:30]: thanks!
